pom.xml

```diff
@@ -271,6 +281,221 @@
                     <artifactId>jacoco-maven-plugin</artifactId>
                     <version>${jacoco.version}</version>
                 </plugin>
+                <plugin>
```

Comment:
What does declaring this as a plugin do here?

Reply:
This is in the `<pluginManagement>` block, so its only to lock down the version, which I did to have the same version anywhere it would be applied as well as the `<annotationProcessorPaths>` when configuring the maven compiler plugin.

Reply:
My understanding is that this is a transitive dependency and not used directly. Why is it important to pin the version here?

Reply:
The `<annotationProcessorPaths>` needs a version defined, as I get `Resolution of annotationProcessorPath dependencies failed: Cannot find version for annotation processor path 'org.eclipse.sisu:org.eclipse.sisu.inject.jar'.` otherwise and it looked like a good idea to pin it here so we can set both to the same version using properties + it's a pattern for other plugins as well. Also, the `sisu-maven-plugin` is actually applied in the `micronaut-maven-enforcer-rules` (sub)project.

Reply:
Ah i see why we need it. Makes sense then.